src/stavenote.js

```diff
@@ -725,6 +739,7 @@ export class StaveNote extends StemmableNote {
   setKeyLine(index, line) {
     this.keyProps[index].line = line;
     this.note_heads[index].setLine(line);
+    this.reset();
```

Comment:
Also worth mentioning that while we could update the notehead individually, I think it's better to use the `.reset()` as a catch-all "something has changed, please rebuild" (which also indicates that formatting is happening before `.format` is called, but that's another story).